### DISCUSSION TITLE
Ubuntu Server 21.10 on RPi4-Cluster - missing "linux-modules-extra-raspi"

### DISCUSSION THREAD

**Discussion by JoeKravelli** [techno-tim/k3s-ansible#25] [Questions] [answered]
I recently provisioned a rpi-cluster with the help of awesome k3s-ansible script, but it failed. The reason was a missing package called "linux-modules-extra-raspi". It contains the essential module "vxlan", which is the default networking backend for flannel (see also: https://rancher.com/docs/k3s/latest/en/installation/network-options/).

Error messages were changing, but mainly saying this:
"The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?"
..or this:
"Error from server (InternalError): an error on the server ("apiserver not ready") has prevented the request from succeeding"

By installing the package with "sudo apt install linux-modules-extra-raspi" on all raspberry pi nodes, rebooting, undeploying and redeploying the k3s-ansible script, all went smoothly.

This is probably a specific case, but by pre-checking, if the module is available, the error could be avoided.
It can be checked by typing "modprobe vxlan". If the output is this:
"modprobe: FATAL: Module vxlan not found in directory /lib/modules/5.13.0-1008-raspi", then the module is missing.

I found this issue here:
https://github.com/k3s-io/k3s/issues/4234

Hoping this will help some folks!
J.K.

**Comment by timothystewart6** [comment, marked as answer]
This has been added to the playbook!  Thank you!

**Comment by depasseg** [comment]
Interesting.  My playbook is failing at this step because this package can't be found on my Ubuntu 20.04 LTS install on a Pi CM4.  I might be mistaken but it doesn't look like that package is available for 20.04 LTS.  (https://packages.ubuntu.com/search?keywords=linux-modules-extra-raspi) I'll try upgrading the distro and see if the ansible playbook completes.
`TASK [raspberrypi : Install linux-modules-extra-raspi] *********************************************************************************************************************************************************************************************************************************************************************
Monday 23 January 2023  19:07:15 -0500 (0:00:00.632)       0:00:12.044 ********
fatal: [172.16.1.148]: FAILED! => changed=false
  msg: No package matching 'linux-modules-extra-raspi' is available
fatal: [172.16.1.120]: FAILED! => changed=false
  msg: No package matching 'linux-modules-extra-raspi' is available
fatal: [172.16.1.249]: FAILED! => changed=false
  msg: No package matching 'linux-modules-extra-raspi' is available
fatal: [172.16.1.216]: FAILED! => changed=false
  msg: No package matching 'linux-modules-extra-raspi' is available

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************************************************************************************
172.16.1.120               : ok=11   changed=0    unreachable=0    failed=1    skipped=11   rescued=0    ignored=0
172.16.1.148               : ok=11   changed=0    unreachable=0    failed=1    skipped=11   rescued=0    ignored=0
172.16.1.216               : ok=11   changed=0    unreachable=0    failed=1    skipped=11   rescued=0    ignored=0
172.16.1.249               : ok=11   changed=0    unreachable=0    failed=1    skipped=11   rescued=0    ignored=0

Monday 23 January 2023  19:07:17 -0500 (0:00:02.003)       0:00:14.048 ********
===============================================================================`

EDITED TO ADD:
Upgrading to 22.04 LTS resolved the error.
